### PR TITLE
Collapse empty columns

### PR DESCRIPTION
**Pull request by Senryoku**
Experimental improvements to the Card Pool UI

### CODE REVIEW

**Comment by Riviertje** [comment]
I like the collapsible columns. From my limited testing it looks to be driven by where there are cards, would it be nice to give some user control there?

**Comment by wrmacrae** [comment]
I really like this concept - it was bothering me watching people stream ZNR practice drafts with the new sideboard on the right because they often had CMCs 5+ or 6+ wrapping below and then had to scroll up and down a lot. I also noticed that causing confusion once or twice. I also think it's nice that the collapsed columns are still a bit present, since that makes it clearer what CMC the columns are. I'd suggest collapsing all empty columns, including ones bracketed by non-empty columns. If I've drafted CMC 5 and CMC 1 cards only, I don't need that much screen space devoted to CMCs 2 through 4.

File: client/src/components/CardPool.vue [hunk 1]
```diff
@@ -182,6 +171,24 @@ export default {
 			this.columns.pop();
 		},
 	},
+	computed: {
+		shouldCollapse: function() {
+			const r = [].fill(this.columns.length, false);
```
Comment (wrmacrae) [inline client/src/components/CardPool.vue:176]:
```suggestion
			return this.columns.map(column => column.length === 0);
```

Comment (wrmacrae) [reply client/src/components/CardPool.vue:176]: (per my comment on the PR, if we want to collapse all empty columns, should be really easy (which I imagine you already know)

**Comment by Senryoku** [comment]
About letting the user control the collapse: I thought of this change because, like @wrmacrae, I've seen people confused by cards wrapping down but never realizing they could remove columns manually. Adding another manual tweak won't help them with that.

Removing wrapping all together and adding horizontal scroll using `overflow-x: scroll` on the card pool div is probably a cleaner solution (provided that horizontal scrolling isn't too problematic; we'd probably have to remap the mouse wheel to horizontal scrolling while the cursor is in the div to make it intuitive), but I couldn't get it to work properly. 

Yes, collapsing all columns is easier, I limited it to both sides mainly because I wasn't happy with the behaviour of the first version. I re-worked it a bit so the transitions are nicer and applied it to all columns, you can try it with the new deployed version.

**Comment by wrmacrae** [comment]
This looks good to me! I'm very interested to see how this works out with different users and resolutions, so I think we should probably merge and keep an eye on usage, and then we can decide on other tweaks from there.

**Pull request merged**